Allow Transport Actions to indicate authN realm (#45767)
elastic/elasticsearch: x-pack/docs/en/rest-api/security/oidc-prepare-authentication-api.asciidoc

The following parameters can be specified in the body of the request:

`realm`::
  (Optional, string) The name of the OpenID Connect realm in {es} the configuration of which should
be used in order to generate the authentication request. Cannot be specified

> when `iss` is specified. One of `realm`, `iss` needs to be specified.

Reviewer: ```suggestion
when `iss` is specified. One of `realm`, `iss` is required.
```

`state`::
  (Optional, string) Value used to maintain state between the authentication request and the
response, typically used as a Cross-Site Request Forgery mitigation. If the
caller of the API doesn't provide a value, {es} will generate one with
sufficient entropy itself and return it in the response.

`nonce`::
  (Optional, string) Value used to associate a Client session with an ID Token and to mitigate
replay attacks. If the caller of the API doesn't provide a value, {es} will
generate one with sufficient entropy itself and return it in the response.

`iss`::
  (Optional, string) In the case of a 3rd Party initiated Single Sign On, this is the Issuer
Identifier for the OP that the RP is to send the Authentication Request to.

> Cannot be specified when `realm` is specified. One of `realm`, `iss` needs to be specified.

Reviewer: ```suggestion
Cannot be specified when `realm` is specified. One of `realm`, `iss` is required.
```
